app/service/file_svc.py

```diff
@@ -169,12 +170,27 @@ def _read(self, filename):
         :param ldflags: A string of ldflags to use when building the go executable
         :param cflags: A string of CFLAGS to pass to the go compiler
         :param buildmode: GO compiler buildmode flag
+        :param build_dir: The path to build should take place in
         :return:
         """
-        os.system(
-            'cd %s && %s %s go build %s -o %s -ldflags=\'%s\' %s' % (build_dir, _go_vars(arch, platform), cflags,
-                                                                     buildmode, output, ldflags, src_fle)
-        )
+        env = copy.copy(os.environ)
+        env['GOARCH'] = arch
+        env['PLATFORM'] = platform
+        if cflags:
+            for cflag in cflags.split(' '):
+                name, value = cflag.split('=')
+                env[name] = value
+
+        build_mode_args = ['-buildmode', buildmode] if buildmode else []
+
+        process = await asyncio.subprocess.create_subprocess_exec('go', 'build', *build_mode_args, '-o', output,
+                                                                  '-ldflags', ldflags, src_fle,
+                                                                  cwd=build_dir, env=env,
+                                                                  stdout=asyncio.subprocess.PIPE,
+                                                                  stderr=asyncio.subprocess.PIPE)
+        command_output = await process.communicate()
```

Comment:
not using this output variable anywhere?

Reply:
https://github.com/mitre/caldera/pull/1495/files#diff-21f6cfa32c5798d28444f10f171e3c40R193